### ISSUE TITLE
Add an option to download all tracks

### ISSUE THREAD

**Issue by Flygrounder** [leinelissen/jellyfin-audio-player#70]
First of all, thanks for that amazing app - it's the best I've seen so far.

Can you add an option to download all audios to listen offline, not just from one album or playlist?

**Comment by leinelissen** [comment]
Thanks! That sounds like a definite possibility, although I agree I would need to implement track view first. There might be some performance issues for rendering larger libraries there, so I need to take a look at how I can make that work. I would also need to take a look at making the downloader a bit more resilient when it has to possibly handle a couple of thousand simultaneous downloads. And of course a huge whopping warning that it's going to destroy your data bill and phone performance ;)

**Comment by Flygrounder** [comment]
Also I would suggest moving download button to the top of the album/playlist view. Because currently you have to scroll through all tracks to find it.

**Comment by leinelissen** [comment]
This is a fair remark! Right now, I've chosen to prioritise the playing button and the tracklist as they are in more continuous use as opposed to the single download event. This is kind of cumbersome with long albums and playlists, so I maybe need to reshuffle some of the ideas around playlist/album interactions. Thanks for the feedback!

**Comment by Flygrounder** [comment]
I've tried bypassing that by creating a playlist with all my tracks (~450). When I click download, app becomes unresponsive, the only thing that works is scrolling and no buttons work. So it seems like downloading needs to be improved as well.

**Comment by leinelissen** [comment]
Yeah this figures, it tries to download all the files at once and this causes so many updates that it blocks the app. That's something that needs to be fixed.

**Comment by Cj1m** [comment]
Have added a PR for this, but note that it does not address the issue that occurs when downloading too many files.

The PR is primarily for viewing all tracks, since singles which were not in albums were not appearing for me.
It seemed relevant to add a "Download All" button to that view.

**Comment by NoSadBeHappy** [comment]
This is a must needed for me. I just want to use jellyfin so sync my MP3s instead of manually doing it, so I use offline mode all the time. It would also be nice if it somehow notified you if your offline music and playlists are not synced with the most recent version on the server.

**Comment by leinelissen** [comment]
Hey thanks for all the input on this, I hear all of you. I'll try to prioritise this whenever I get around to writing some code, but I have a full schedule for the foreseeable future.
